Library/Homebrew/commands.rb

```diff
@@ -194,7 +196,35 @@ def rebuild_commands_completion_list
     # Ensure that the cache exists so we can build the commands list
     HOMEBREW_CACHE.mkpath
 
+    cmds = commands(aliases: true).reject do |cmd|
+      # TODO: remove the cask check when `brew cask` is removed
+      cmd.start_with?("cask ") || Completions::COMPLETIONS_EXCLUSION_LIST.include?(cmd)
+    end
+
     file = HOMEBREW_CACHE/"all_commands_list.txt"
-    file.atomic_write("#{commands(aliases: true).sort.join("\n")}\n")
+    file.atomic_write("#{cmds.sort.join("\n")}\n")
+  end
+
+  def command_options(command)
+    path = Commands.path(command)
+    return unless path
+
+    if cmd_parser = Homebrew::CLI::Parser.from_cmd_path(path)
+      cmd_parser.processed_options.map do |short, long, _, desc|
+        [long || short, desc]
+      end
+    else
+      options = []
+      comment_lines = path.read.lines.grep(/^#:/)
+      return options if comment_lines.empty?
+
+      # skip the comment's initial usage summary lines
+      comment_lines.slice(2..-1).each do |line|
+        if / (?<option>-[-\w]+) +(?<desc>.*)$/ =~ line
+          options << [option, desc]
```

Comment:
I know you didn't originally write this, but I didn't know that we could declare variables from regex(!)